policy-controller/k8s/index/src/outbound/index.rs

```diff
@@ -730,51 +884,53 @@ fn is_service(group: Option<&str>, kind: &str) -> bool {
 impl ServiceRoutes {
     fn watch_for_ns_or_default(&mut self, namespace: String) -> &mut RoutesWatch {
         // The routes from the producer namespace apply to watches in all
-        // namespaces so we copy them.
+        // namespaces, so we copy them.
         let routes = self
             .watches_by_ns
             .get(self.namespace.as_ref())
             .map(|watch| watch.routes.clone())
-            .unwrap_or_default();
+            .unwrap_or_else(|| None);
+
         self.watches_by_ns.entry(namespace).or_insert_with(|| {
             let (sender, _) = watch::channel(OutboundPolicy {
-                http_routes: Default::default(),
-                authority: self.authority.clone(),
-                name: self.name.to_string(),
-                namespace: self.namespace.to_string(),
                 port: self.port,
                 opaque: self.opaque,
                 accrual: self.accrual,
+                routes: routes.clone(),
+                name: self.name.to_string(),
+                authority: self.authority.clone(),
+                namespace: self.namespace.to_string(),
             });
+
             RoutesWatch {
-                opaque: self.opaque,
-                accrual: self.accrual,
                 routes,
                 watch: sender,
+                opaque: self.opaque,
+                accrual: self.accrual,
             }
         })
     }
 
-    fn apply(&mut self, gknn: GroupKindNamespaceName, route: HttpRoute) {
+    fn apply(&mut self, gknn: GroupKindNamespaceName, route: TypedOutboundRoute) {
         if *gknn.namespace == *self.namespace {
             // This is a producer namespace route.
             let watch = self.watch_for_ns_or_default(gknn.namespace.to_string());
-            watch.routes.insert(gknn.clone(), route.clone());
-            watch.send_if_modified();
+
+            watch.insert_route(gknn.clone(), route.clone());
+
             // Producer routes apply to clients in all namespaces, so
             // apply it to watches for all other namespaces too.
-            for (ns, watch) in self.watches_by_ns.iter_mut() {
-                if ns != &gknn.namespace {
-                    watch.routes.insert(gknn.clone(), route.clone());
-                    watch.send_if_modified();
-                }
-            }
+            self.watches_by_ns
```

Comment:
is this a functional or stylistic change?

Reply:
Mmmmm... _technically_ both? I moved the `send_if_modified` calls into the methods that actually do the mutation that would necessitate updates being sent to guard against the possibility of future changes adding mutations but forgetting to call `send_if_modified`. The style change is just tidying up after the fact